### ISSUE TITLE
[Bug]: UI crash when `verifyingContract` is `undefined`

### ISSUE THREAD

**Issue by Gudahtt** [MetaMask/metamask-extension#26910]
### Describe the bug

When `signTypedData_v4` Permit signature is called with an undefined `verifyingContract`, the UI crashes.

### Expected behavior

There should be no UI crash.

### Screenshots/Recordings

https://github.com/user-attachments/assets/0f94ffc8-8c07-4d6c-8adc-8890c71c53a1

### Steps to reproduce

* Connect to a dapp
* Ensure that Ethereum Mainnet is the selected chain for that dapp
* Run this in the console:
```
await window.ethereum.request({
  method: 'eth_signTypedData_v4',
  params: [
    window.ethereum.selectedAddress,
     `{"types":{"EIP712Domain":[{"name":"name","type":"string"},{"name":"version","type":"string"},{"name":"chainId","type":"uint256"}],"Permit":[{"name":"owner","type":"address"},{"name":"spender","type":"address"},{"name":"value","type":"uint256"},{"name":"nonce","type":"uint256"},{"name":"deadline","type":"uint256"}]},"primaryType":"Permit","domain":{"name":"USD Coin","chainId":${1},"version":"2"},"message":{"owner":"${window.ethereum.selectedAddress}","spender":"0x1661F1B207629e4F385DA89cFF535C8E5Eb23Ee3","value":"1033366316628","nonce":1,"deadline":1678709555}}`,
  ],
})
```

### Detection stage

In production (default)

### Version

12.1.0

### Build type

None

### Browser

Chrome

### Operating system

MacOS

**Comment by Gudahtt** [comment]
Caused by this line: https://github.com/MetaMask/metamask-extension/blob/1998dc46ed0df1f6085a73ebd63de5ca74ff3eba/ui/pages/confirmations/components/confirm/info/typed-sign/typed-sign.tsx#L81

**Comment by Gudahtt** [comment]
PR description upgraded and severity downgraded to Sev2, because we discovered that this only applies to Permit signatures without a `verifyingContract`. But the Permit spec requires a `verifyingContract`.

**Comment by bschorchit** [comment]
This crash is fixed on v12.2
